__tests__/integration/component-slots/index.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  component,
+  staticNode,
+  slotNode,
+  elementNode,
+} from '../../../src/shared/builders/uidl-builders'
+
+import { createReactComponentGenerator, createVueComponentGenerator } from '../../../src'
+
+const reactGenerator = createReactComponentGenerator()
+const vueGenerator = createVueComponentGenerator()
+
+describe('Component Slot Node', () => {
+  describe('Simple Slot', () => {
+    const uidl = component('Simple Slot', elementNode('container', {}, [slotNode()]))
+
+    it('renders props.children in React', async () => {
+      const result = await reactGenerator.generateComponent(uidl)
+      const code = result.files[0].content
+
+      expect(code).toContain('{props.children}')
+    })
+
+    it('renders a <slot> tag in Vue', async () => {
+      const result = await vueGenerator.generateComponent(uidl)
+      const code = result.files[0].content
+
+      expect(code).toContain('<slot></slot>')
+    })
+  })
+
+  describe('Slot with fallback', () => {
+    const uidl = component(
+      'Slot Component',
+      elementNode('container', {}, [slotNode(elementNode('text', {}, [staticNode('Placeholder')]))])
+    )
+
+    it('renders props.children and placeholder in React', async () => {
+      const result = await reactGenerator.generateComponent(uidl)
+      const code = result.files[0].content
+
+      expect(code).toContain('{props.children || <span>Placeholder</span>}')
+    })
+
+    it('renders a <slot> tag and placeholder in Vue', async () => {
+      const result = await vueGenerator.generateComponent(uidl)
+      const code = result.files[0].content
+
+      expect(code).toContain('<slot><span>Placeholder</span></slot>')
+    })
+  })
+
+  describe('Slot with name', () => {
+    const uidl = component(
+      'Slot Component',
+      elementNode('container', {}, [slotNode(undefined, 'slot-1')])
+    )
+
+    it('renders props.children in React', async () => {
+      const result = await reactGenerator.generateComponent(uidl)
+      const code = result.files[0].content
+
+      expect(code).toContain('{props.children}')
```

Comment:
If `slot-1` is a named slot, the the expected props to be sent over to react would  be `props['slot-1']`. Otherwise react would not have feature parity with vue. Vue will be able to have header, footer and main content as 3 separate slots configurable, while react will only support one.

Components like layouts (header, content, footer) need this in my opinion.

Reply:
I was wondering if we need some sort of abstraction for the slots when passing them.

In Vue, there's a standard where you add `v-slot:header` `v-slot:footer` as attributes, so we wouldn't be able to use the same UIDL for vue/react if we don't have some sort of abstraction on that end